docs/_posts/examples/3400-01-24-query-similar-features.html

```diff
@@ -0,0 +1,137 @@
+var popup = new mapboxgl.Popup({
+    closeButton: false
+});
+
+function renderOverlay(features) {
+    // Clear any overlay content
+    overlay.innerHTML = '';
+
+    if (features.length) {
+        // Total the population of all features
+        var populationSum = features.reduce(function(memo, feature) {
+            return memo + feature.properties.population;
+        }, 0);
+
+        var title = document.createElement('strong');
+        title.textContent = features[0].properties.COUNTY + ' (' + features.length + ' found)';
+
+        var population = document.createElement('div');
+        population.textContent = 'Total population: ' + populationSum.toLocaleString();
+
+        overlay.appendChild(title);
+        overlay.appendChild(population);
+        overlay.style.display = 'block';
+    } else {
+        overlay.style.display = 'none';
+    }
+}
+
+map.on('load', function() {
+    var canvas = map.getCanvasContainer();
+
+    // Add the source to query. In this example we're using
+    // county polygons uploaded as vector tiles
+    map.addSource('counties', {
+        "type": "vector",
+        "url": "mapbox://mapbox.82pkq93d"
+    });
+
+    map.addLayer({
+        "id": "counties",
+        "type": "fill",
+        "source": "counties",
+        "source-layer": "original",
+        "paint": {
+            "fill-outline-color": "rgba(0,0,0,0.1)",
+            "fill-color": "rgba(0,0,0,0.1)"
+        }
+    }, 'place-city-sm'); // Place polygon under these labels.
+
+    map.addLayer({
+        "id": "counties-highlighted",
+        "type": "fill",
+        "source": "counties",
+        "source-layer": "original",
+        "paint": {
+            "fill-outline-color": "#484896",
+            "fill-color": "#6e599f",
+            "fill-opacity": 0.75
+        },
+        "filter": ["in", "COUNTY", ""]
+    }, 'place-city-sm'); // Place polygon under these labels.
+
+    map.on('mousemove', function(e) {
+        var features = map.queryRenderedFeatures(e.point, {
+            layers: ['counties']
+        });
+
+        // Change the cursor style as a UI indicator.
+        map.getCanvas().style.cursor = (features.length) ? 'pointer' : '';
+
+        // Remove things if no feature was found.
+        if (!features.length) {
+            popup.remove();
+            map.setFilter('counties-highlighted', ['in', 'COUNTY', '']);
+            renderOverlay([]);
+            return;
+        }
+
+        // Single out the first found feature on mouseove.
+        var feature = features[0];
+
+        // Query the counties layer visible in the map. Use the filter
+        // param to only collect results that share the same county name.
+        var relatedFeatures = map.querySourceFeatures('counties', {
+            sourceLayer: 'original',
+            filter: ['in', 'COUNTY', feature.properties.COUNTY]
+        });
+
+        // Render found features in an overlay.
+        renderOverlay(relatedFeatures);
```

Comment:
I'm following now :) inline the code I created a named function for. It only happens in one place.